post_deployment_tests/deployment_test.py

```diff
@@ -0,0 +1,338 @@
+
+        f.close()
+        self.archive_files[fileid] = fileid
+        self.assertEqual(resp.status_code, 201)
+        data = resp.json()
+        self.assertEqual(int(data['total_bytes']), self.filesize['size'])
+        self.assertEqual(data['message'], 'File added to archive')
+
+    def test_large_binary_file_status(self):
+        """test statusing a large binary file."""
+        fileid = '9999'
+        resp = requests.head(str(ARCHIVEURL + fileid))
+        self.assertEqual(resp.status_code, 204)
+        self.assertEqual(resp.headers['x-pacifica-file-storage-media'], 'disk')
+        self.assertEqual(resp.headers['content-length'], str(self.filesize['size']))
+        self.assertEqual(resp.headers['x-pacifica-messsage'], 'File was found')
+
+    def test_large_binary_file_stage(self):
+        """test staging a large binary file."""
+        fileid = '9999'
+        resp = requests.post(str(ARCHIVEURL + fileid))
+        self.assertEqual(resp.status_code, 200)
+        data = resp.json()
+        self.assertEqual(data['message'], 'File was staged')
+
+    def test_large_binary_file_read(self):
+        """test reading a large binary file."""
+        fileid = '9999'
+        filename = '/tmp/test_binary_read'
+        resp = requests.get(str(ARCHIVEURL + fileid), stream=True)
+        myfile = open(filename, 'wb+')
+        buf = resp.raw.read(1024)
+        while buf:
+            myfile.write(buf)
+            buf = resp.raw.read(1024)
+        myfile.close()
+        self.local_files[filename] = filename
+        filesize = os.path.getsize(filename)
+        self.assertEqual(filesize, self.filesize['size'])
+
+    def test_large_binary_cleanup(self):
+        """Clean up files that are created for testing."""
+        if CLEANLOCALFILES:
+            for filepath in self.local_files:
+                os.remove(filepath)
+
+        if CLEANARCHIVEFILES:
+            for filepath in self.archive_files:
+                os.remove(ARCHIVEPREFIX + filepath)
+        self.assertEqual(True, True)
+
+
+class ManyFileArchiveTests(unittest.TestCase):
+    """Class that tests the writing of many files at once."""
+
+    local_files = {}
+    archive_files = {}
+
+    def test_many_file_write(self):
+        """test writing many files to the archive."""
+        for i in range(3000, 4000):
+            filename = '/tmp/test_simple_write' + str(i) + '.txt'
+            fileid = str(i)
+            file1 = open(filename, 'w+')
+            file1.write('Writing content for first file')
+            file1.close()
+            self.local_files[filename] = filename
+            f = open(filename, 'rb')
+            resp = requests.put(str(ARCHIVEURL + fileid), data=f)
+            f.close()
+            self.assertEqual(resp.status_code, 201)
+            self.archive_files[fileid] = fileid
+            data = resp.json()
+            self.assertEqual(data['message'], 'File added to archive')
+
+    def test_many_file_cleanup(self):
```

Comment:
If you use the `tempfile` module correctly the files would disappear when the script exits.